src/Surfnet/StepupGateway/GatewayBundle/Saml/ResponseContext.php

```diff
@@ -162,7 +167,9 @@ public function saveAssertion(SAML2_Assertion $assertion)
 
         // same for the entityId of the authenticating Authority
         $authenticatingAuthorities = $assertion->getAuthenticatingAuthority();
-        if (!empty($authenticatingAuthorities)) {
+        if (empty($authenticatingAuthorities)) {
+            $this->stateHandler->setAuthenticatingIdp($assertion->getIssuer());
+        } else {
             $this->stateHandler->setAuthenticatingIdp(reset($authenticatingAuthorities));
```

Comment:
I added this because SimpleSAMLphp doesn't include any authenticating authorities. Should we fall back to the message issuer or should IdPs support authenticating authorities? If they don't, they circumvent SP/IdP LoA requirements.